--- a/cli/src/commands/commit.rs
+++ b/cli/src/commands/commit.rs
@@ -29,6 +29,6 @@ use crate::ui::Ui;
 pub(crate) struct CommitArgs {
-    /// Interactively choose which changes to include in the first commit
+    /// Commit all changes
     #[arg(short, long)]
-    interactive: bool,
-    /// Specify diff editor to be used (implies --interactive)
+    all: bool,
+    /// Specify diff editor to be used
     #[arg(long, value_name = "NAME")]
@@ -70,3 +70,3 @@ pub(crate) fn cmd_commit(
     let diff_selector =
-        workspace_command.diff_selector(ui, args.tool.as_deref(), args.interactive)?;
+        workspace_command.diff_selector(ui, args.tool.as_deref(), !args.all)?;
     let mut tx = workspace_command.start_transaction();
