--- a/src/polkitmateauthenticationdialog.c
+++ b/src/polkitmateauthenticationdialog.c
@@ -692,5 +692,5 @@ polkit_mate_authentication_dialog_constructed (GObject *object)
   /* main message */
   label = gtk_label_new (NULL);
-  s = g_strdup_printf ("<big><b>%s</b></big>", dialog->priv->message);
+  s = g_strdup_printf ("<big><b>%s</b></big>", "Authentication is needed to run this program.");
   gtk_label_set_markup (GTK_LABEL (label), s);
   g_free (s);
@@ -699,5 +699,5 @@ polkit_mate_authentication_dialog_constructed (GObject *object)
   gtk_label_set_yalign (GTK_LABEL (label), 0.5);
   gtk_label_set_line_wrap (GTK_LABEL (label), TRUE);
-  gtk_label_set_max_width_chars (GTK_LABEL (label), 50);
+  gtk_label_set_max_width_chars (GTK_LABEL (label), 80);
   gtk_box_pack_start (GTK_BOX (main_vbox), label, FALSE, FALSE, 0);
 
@@ -730,5 +730,5 @@ polkit_mate_authentication_dialog_constructed (GObject *object)
   gtk_label_set_line_wrap (GTK_LABEL (label), TRUE);
   gtk_label_set_max_width_chars (GTK_LABEL (label), 50);
-  gtk_box_pack_start (GTK_BOX (main_vbox), label, FALSE, FALSE, 0);
+  // gtk_box_pack_start (GTK_BOX (main_vbox), label, FALSE, FALSE, 0);
 
   /* user combobox */
@@ -788,6 +788,20 @@ polkit_mate_authentication_dialog_constructed (GObject *object)
   gtk_box_pack_start (GTK_BOX (details_vbox), grid, FALSE, FALSE, 0);
 
+  label = gtk_label_new (NULL);
+  s = g_strdup_printf ("<small>%s</small>", dialog->priv->message);
+  gtk_label_set_markup (GTK_LABEL (label), s);
+  g_free (s);
+
+  gtk_label_set_xalign (GTK_LABEL (label), 0.0);
+  gtk_label_set_yalign (GTK_LABEL (label), 1.0);
+
+  s = g_strdup_printf ("<small><b>%s:</b></small>", "Full message");
+  add_row (grid, rows, s, label);
+  g_free (s);
+
+  rows++;
+
   /* TODO: sort keys? */
-  rows = 0;
+  rows = 1;
   if (dialog->priv->details != NULL)
     {
