--- a/libnemo-private/nemo-icon-info.c
+++ b/libnemo-private/nemo-icon-info.c
@@ -668,2 +668,0 @@
-        return NEMO_LIST_ICON_SIZE_SMALLEST;
-    case NEMO_ZOOM_LEVEL_SMALL:
@@ -671,1 +669,1 @@
-    case NEMO_ZOOM_LEVEL_STANDARD:
+    case NEMO_ZOOM_LEVEL_SMALL:
@@ -673,1 +671,1 @@
-    case NEMO_ZOOM_LEVEL_LARGE:
+    case NEMO_ZOOM_LEVEL_STANDARD:
@@ -675,1 +673,1 @@
-    case NEMO_ZOOM_LEVEL_LARGER:
+    case NEMO_ZOOM_LEVEL_LARGE:
@@ -677,1 +675,1 @@
-    case NEMO_ZOOM_LEVEL_LARGEST:
+    case NEMO_ZOOM_LEVEL_LARGER:
@@ -679,0 +677,2 @@
+    case NEMO_ZOOM_LEVEL_LARGEST:
+        return NEMO_LIST_ICON_SIZE_LARGEST;
--- a/libnemo-private/nemo-icon-info.h
+++ b/libnemo-private/nemo-icon-info.h
@@ -36,6 +36,6 @@
-#define NEMO_LIST_ICON_SIZE_SMALLER  24
-#define NEMO_LIST_ICON_SIZE_SMALL    32
-#define NEMO_LIST_ICON_SIZE_STANDARD 48
-#define NEMO_LIST_ICON_SIZE_LARGE    72
-#define NEMO_LIST_ICON_SIZE_LARGER   96
-#define NEMO_LIST_ICON_SIZE_LARGEST  192
+#define NEMO_LIST_ICON_SIZE_SMALLER  18
+#define NEMO_LIST_ICON_SIZE_SMALL    24
+#define NEMO_LIST_ICON_SIZE_STANDARD 32
+#define NEMO_LIST_ICON_SIZE_LARGE    48
+#define NEMO_LIST_ICON_SIZE_LARGER   72
+#define NEMO_LIST_ICON_SIZE_LARGEST  96
