--- a/Main.qml
+++ b/Main.qml
@@ -84,7 +84,7 @@
                     font.pointSize: root.generalFontSize
                     renderType: Text.QtRendering
                     function updateTime() {
-                        text = new Date().toLocaleString(Qt.locale("en_US"), "ddd dd MMMM,  hh:mm A")
+                        text = new Date().toLocaleString(Qt.locale("en_DK"), "dd/MM/yyyy  hh:mm")
                     }
                 }
                 Timer {
