--- a/src/programs/pkexec.c
+++ b/src/programs/pkexec.c
@@ -833,15 +833,22 @@ main (int argc, char *argv[])
   polkit_details_insert (details, "program", path);
   polkit_details_insert (details, "command_line", command_line);
 
-  cmdline_short = g_strdup(command_line);
-  if (cmdline_short == NULL)
-    {
-      g_printerr ("Failed to allocate memory for shortened command line.\n");
-      goto out;
+  gchar **cmdline_tokens = g_strsplit(command_line, " ", 0);
+  cmdline_short = g_path_get_basename(cmdline_tokens[0]);
+  if (g_strcmp0(cmdline_short, "env") == 0) {
+    guint token_index;
+    for (token_index = 1; cmdline_tokens[token_index] != NULL; token_index++) {
+      if (strchr(cmdline_tokens[token_index], '=') == NULL)
+        break;
     }
-  if (strlen(command_line) > 80)
-      g_stpcpy(g_stpcpy( cmdline_short + 38, " ... " ),
-               command_line + strlen(command_line) - 37 );
+    g_free(cmdline_short);
+    if (cmdline_tokens[token_index] != NULL)
+      cmdline_short = g_path_get_basename(cmdline_tokens[token_index]);
+    else
+      cmdline_short = g_strdup(command_line);
+  }
+  g_strfreev(cmdline_tokens);
+
   polkit_details_insert (details, "cmdline_short", cmdline_short);
 
   if (g_strcmp0 (action_id, "org.freedesktop.policykit.exec") == 0)
